### PR TITLE
add LoongArch64 ELF basic support

### PR DESCRIPTION
**Pull request by xry111**
The LoongArch architecture (LoongArch) is an Instruction Set
Architecture (ISA) that has a Reduced Instruction Set Computer (RISC)
style.
The documents are on
https://loongson.github.io/LoongArch-Documentation/README-EN.html

The ELF ABI Documents are on:
https://loongson.github.io/LoongArch-Documentation/LoongArch-ELF-ABI-EN.html

### CODE REVIEW

File: src/common.rs [hunk 1]
```diff
@@ -23,6 +23,7 @@ pub enum Architecture {
     S390x,
     Sparc64,
     Wasm32,
+    LoongArch64,
```
Comment (bjorn3) [inline src/common.rs:26]: This enum is (mostly) sorted. Can you move this variant to the appropriate location?

Comment (xry111) [reply src/common.rs]: Done.

Comment (xry111) [reply src/common.rs]: Sorry, forgot to sort the `match` expression in `address_size`.  Fixed now.

**Comment by philipc** [comment]
Please add this architecture to the [`elf_any` test](https://github.com/gimli-rs/object/blob/389c43a7567df94bbb31c697f79ed473523a7089/tests/round_trip/mod.rs#L231). You'll probably need to implement a bit more to get this passing.

File: src/read/elf/file.rs [hunk 2]
```diff
@@ -175,6 +175,7 @@ where
             // We only support the 64-bit variant s390x here.
             (elf::EM_S390, true) => Architecture::S390x,
             (elf::EM_SPARCV9, true) => Architecture::Sparc64,
+            (elf::EM_LOONGARCH, true) => Architecture::LoongArch64,
```
Comment (philipc) [inline src/read/elf/file.rs:178]: Please move this before `Architecture::Mips`.

File: src/write/elf/object.rs [hunk 3]
```diff
@@ -83,6 +83,7 @@ impl<'a> Object<'a> {
             Architecture::Riscv32 => true,
             Architecture::S390x => true,
             Architecture::Sparc64 => true,
+            Architecture::LoongArch64 => true,
```
Comment (philipc) [inline src/write/elf/object.rs:86]: Please move this before `Architecture::Mips`.

File: src/write/elf/object.rs [hunk 4]
```diff
@@ -279,6 +280,7 @@ impl<'a> Object<'a> {
             Architecture::Riscv64 => elf::EM_RISCV,
             Architecture::S390x => elf::EM_S390,
             Architecture::Sparc64 => elf::EM_SPARCV9,
+            Architecture::LoongArch64 => elf::EM_LOONGARCH,
```
Comment (philipc) [inline src/write/elf/object.rs:283]: Please move this before `Architecture::Mips`.

**Review by philipc — CHANGES_REQUESTED** [review]

File: src/write/elf/object.rs (hunk 4 above)
Comment (xry111) [reply src/write/elf/object.rs]: Done.

**Pull request merged**